appeals/web/src/common/conjoined-promises.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * @template Arg
+ * @template TResolution
+ * @param {Arg[]} argArr
+ * @param {(arg0: Arg) => Promise<TResolution>} asyncFunc
+ * @returns {Promise<Map<Arg, TResolution>>}
+ */
+export const conjoinedPromises = async (argArr, asyncFunc) => {
```

Comment:
I'm struggling to understand how this function works and what it's intended to do. The TSDoc is pretty mind-bending for anyone without advanced typescript knowledge, and the code in the body also seems difficult to follow, for me personally anyway. Is this really necessary to support the submappers approach, and if so is there anything we could do to make it easier to understand? Some descriptive comments would help a lot. I'm also a bit concerned about the use of `Map` as we haven't used that anywhere else as far as I'm aware. That in itself isn't a good reason not to use it, but it all adds extra cognitive overhead for whoever is extending or maintaining this in future, and I don't understand why the `resolutionMap` can't just be a regular javascript object. It might also be worth renaming the function as it's unclear what's meant by the word `conjoined` in this context.